apps/snitch_core/test/data/model/review_test.exs

```diff
@@ -5,58 +5,52 @@ defmodule Snitch.Data.Model.ReviewTest do
   import Snitch.Factory
   alias Snitch.Data.Model.{Review, ProductReview}
 
-  setup do
-    product = insert(:product)
-    [product: product]
+  setup :reviews
+
+  @tag review_count: 2
+  describe "update/2" do
+    test "updates a review successfully", %{reviews: reviews} do
+      [review, review_2] = reviews
+      rpv = review_2.rating_option_vote
+      rating_option_vote = Map.from_struct(rpv)
+
```

Comment:
You are making use of rating option vote from another review. Instead, just insert a rating_option_vote and use that.

Reply:
@arjun289 If we try to insert a rating option separately then we will get an error regarding the uniqueness of rating option as we are trying to insert same rating option twice.

Reply:
@Jeshan7 I understand that but, here the issues is rationg_option_vote and that can be inserted right?

Reply:
@arjun289 Ultimately rating_option_vote is nothing but  a map of rating_option as defined in the factory . So inserting rating_option_vote is same as inserting a rating_option.